### ISSUE TITLE
Issue in building in Cygwin 

### ISSUE THREAD

**Issue by rajranjan346** [cisco/libsrtp#153]
I tried to build "libsrtp-2.0.0" code in cygwin but it threw warning and asked to report so I am posting this. Following warning I got
configure: WARNING: winsock2.h: present but cannot be compiled
configure: WARNING: winsock2.h:     check for missing prerequisite headers?
configure: WARNING: winsock2.h: see the Autoconf documentation
configure: WARNING: winsock2.h:     section "Present But Cannot Be Compiled"
configure: WARNING: winsock2.h: proceeding with the compiler's result
configure: WARNING:     ## ------------------------------------------------------ ##
configure: WARNING:     ## Report this to https://github.com/cisco/libsrtp/issues ##
configure: WARNING:     ## ------------------------------------------------------ ##
checking for winsock2.h... no

Also If I tried to build .sln (visual studio project) It is not able to compile. Did I miss something?

**Comment by pabuhler** [comment]
Hi I did not even get that far when I tried.
If you would like to submit patches to get cygwin support working again I would be happy to test.
The visual studio solution and project was updated recently so they should be working now.

**Comment by itollefsen** [comment]
There should be a config.log file in the same directory you ran the configure script.

That file should contain more details about the failure. Note it won't be at the very end; configure will log the defines and headers in use as well as the small sample program after it logged the failure.

So either find the failure message and report it here with a few lines above and beyond (at your own discretion), or upload the entire config.log file.

Note that while it's been a while since I used Cygwin, I was able to make it compile and run, with OpenSSL support, using the mingw64 toolchain under Cygwin. My difficulties were with the Cygwin setup itself (specifically the alternatives system when using mingw), not libsrtp.
